C24_WMDE_Mobile_DE_11

Sperling-0 opened the pull request:
Ticket: https://phabricator.wikimedia.org/T380854

Sperling-0 committed:
Prepare campaign for C24_WMDE_Mobile_DE_11

Ticket: https://phabricator.wikimedia.org/T380854

Sperling-0 committed:
Prepare campaign for C24_WMDE_Mobile_DE_11

Ticket: https://phabricator.wikimedia.org/T380854

Sperling-0 committed:
Implement VAR for C24_WMDE_Mobile_DE_11

We wanted to optimize the markup for the close button, but ran into
issues where the test code depended on the class `wmde-banner-mini-close-button` being present.
We added back the class to make the tests pass, but will optimize our
tests and markup. Tracked in #645

Ticket: https://phabricator.wikimedia.org/T380854

Sperling-0 committed:
Prepare CTRL for C24_WMDE_Mobile_DE_11

- Promote form component from previous banner (amounts for selection can be passed in) to a form component in the correct place
- Clean up 'Minimised banner' leftover files

Ticket: https://phabricator.wikimedia.org/T380854

Sperling-0 committed:
Prepare VAR for C24_WMDE_Mobile_DE_11

Ticket: https://phabricator.wikimedia.org/T380854

gbirke commented:
Just a general comment: @Sperling-0 and me have been pairing on this and we're deliberately editing the `Mikings` theme to have a fast feedback cycle (and because it was not clear at the start that the changes for the new design involve more than just colors). We will probably turn the edited `Mikings` theme into a new theme and then reset `Mikings`.

Abban commented:
Yep, this is a case where it's a whole new theme. You could ask the PM to name it too.

Abban commented on src/themes/Mikings/DonationForm/MultiStepDonation.scss:
```diff
@@ -19,15 +19,17 @@
-			color: var( --donation-form-button-color );
-			background: var( --donation-form-button-background );
+			font-size: 18px;
+			color: black;
```
It may seem like it's faster to add the colours inline here, but making them vars from the outset will save time and headaches in the long run. Going through a theme and replacing existing colours with vars isn't a fun job.

Abban commented on src/themes/Mikings/swatches/color_light.scss:
```diff
@@ -9,6 +9,7 @@ $grey800: #202122;
 $grey700: #54595d;
 $grey400: #a2a9b1;
 $grey100: #eaecf0;
+$grey: #eeeeee;
```
These variables are named using the [semantic colour system](https://dev.to/ynab/a-semantic-color-system-the-theory-hk7) please stick to that. When first defined the vars were incremented by 100s because that gives room to insert new colours between existing ones so don't be afraid to add new ones between as a half step (eg $grey150).

Abban committed:
Add CSS fixups for C24_WMDE_Mobile_DE_11

- Fix imports in VAR banner styles
- Add 2 stage heights for mini banner
- Remove border of mini banner and soft close
- Match paddings and margins up closer to the design
- Force soft close buttons into position with CSS grid
- Remove bold from form legend text weight

Ticket: https://phabricator.wikimedia.org/T380854

Abban committed:
Fix iOS scrolling issue

This adds a mobile Safari only rule that takes
the address bar into account when calculating
viewport height. This stops the bottom of our
modals getting cut off.

It also emits events to tell the page class to
fix the body element and stop page scrolling behind
a modal.

Ticket: https://phabricator.wikimedia.org/T380854

Abban commented:
My suggestion for getting this passing and merged is to move the data attribute commit into a new PR, and change the close button class that was causing the issue during testing to match the other banners.
